src/App.tsx

```diff
@@ -15,36 +21,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_ALPHABETICALLY = 'is-info';
+const SORT_LENGTH = 'is-success';
+
+function sortGoods(sortBy: string, isReversed: boolean) {
+  const sortArr = [...goodsFromServer];
+
+  if (sortBy) {
+    sortArr.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SORT_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortArr.reverse();
+  }
+
+  return sortArr;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<string>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const currentGoods = sortGoods(sortBy, isReversed);
+
+  function reset() {
+    setSortBy('');
+    setIsReversed(false);
+  }
+
+  function changeReversed() {
+    setIsReversed(!isReversed);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortBy(SORT_ALPHABETICALLY)}
+          type="button"
+          className={`button is-info ${sortBy !== SORT_ALPHABETICALLY && 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortBy(SORT_LENGTH)}
+          type="button"
+          className={`button is-success ${sortBy !== SORT_LENGTH && 'is-light'}`}
```

Comment:
Similar to the previous comment, the active button for 'Sort by length' also needs to be visually distinct to comply with the [UX] checklist item. Adjust the className to reflect the active state when `sortBy` matches `SORT_LENGTH`.